src/pages/books.js

```diff
@@ -1,27 +1,30 @@
-import React, { useState } from 'react';
-import BookList from '../components/bookList';
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { addBook, removeBook } from '../redux/books/bookSlice';
+import BookDisplay from '../components/booksDisplay';
 import BookForm from '../components/bookForm';
 
 function Books() {
-  const [books, setBooks] = useState([
-    { id: 1, title: 'Book 1', author: 'Author 1' },
-    { id: 2, title: 'Book 2', author: 'Author 2' },
-  ]);
+  const books = useSelector((state) => state.books.books);
+  const dispatch = useDispatch();
 
-  const handleDeleteBook = (bookId) => {
-    const updatedBooks = books.filter((book) => book.id !== bookId);
-    setBooks(updatedBooks);
+  const handleAddBook = (newBook) => {
+    dispatch(addBook(newBook));
   };
 
-  const handleAddBook = (newBook) => {
-    setBooks([...books, { ...newBook, id: Date.now() }]);
+  const handleDeleteBook = (itemId) => {
+    const bookToDelete = books.find((book) => book.item_id === itemId);
+
+    if (bookToDelete) {
+      dispatch(removeBook(itemId));
+    }
   };
```

Comment:
> - [ ]  Well done for addressing the previous reviewer's request. However, I've observed that when I add a new book and attempt to delete a specific book that was added, it results in the removal of all the newly added books. Kindly address this issue as it could lead to undesirable outcomes.👍
> 
> To better understand the problem, please refer to the attached video here:
> 
> [screen-capture.webm](https://github.com/richie1988/My_Bookstore_App/assets/40009354/dcd1bc58-8490-4423-a9df-7f76b1c298de)
>